# Operasi Matematika & Numpy

# Santi Oktaviani

In [1]:
pip install --upgrade numpy

In [2]:
import numpy as np

### Pengenalan array

###Latihan-2 (2D array)

In [27]:
#2D Array (Matriks)
A = np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
print(A)

[[1 2 3]
 [4 5 6]
 [7 8 9]]


In [28]:
#Lengkapi kodingan berikut
print('banyaknya entry = ', A.size)
print('dimensi array = ', A.ndim)
print('ukuran array = ', A.shape)

banyaknya entry =  9
dimensi array =  2
ukuran array =  (3, 3)


###Latihan-3 (Slicing)

In [29]:
#Lengkapi kodingan berikut
print('baris ke-3 = ', A[2])
print('entry ke-1 & 2 baris ke-3 = ', A[2, :2])
print('entry ke-2 & 3 kolom ke-3 = ', A[1:3, 2])

baris ke-3 =  [7 8 9]
entry ke-1 & 2 baris ke-3 =  [7 8]
entry ke-2 & 3 kolom ke-3 =  [6 9]


output jawaban:
baris ke-3 =  [7 8 9]

entry ke-1 & 2 baris ke-3 =  [7 8]

entry ke-2 & 3 kolom ke-3 =  [6 9]

### Latihan-4 (Operasi matriks)

In [32]:
A = np.array([[1, 2], [7,8]])
B = np.array([[3, 5], [1,6]])

Hitunglah:
*   2*A
*   2+A
*   A+B
*   A-B
*   determinan A
*   invers dari B, simpan jadi matriks D
*   transpose D
*   nilai eigen dan vektor eigen dari A dan B







In [33]:
#Ketik Kode disini
X = 2*A
print("Hasil Perkalian 2*A adalah: ")
print(X)

X = 2+A
print("\nHasil Penjumlahan 2+A adalah: ")
print(X)

X = A+B
print("\nHasil Penjumlahan A+B adalah: ")
print(X)

X = A-B
print("\nHasil Pengurangan A-B adalah: ")
print(X)

print('\nDeterminan matriks A', np.linalg.det(A))

D = np.linalg.inv(B)
print("\nInvers Matriks B: \n", D)

D_t = D.T
print("\nTranspose D: \n", D_t)



Hasil Perkalian 2*A adalah: 
[[ 2  4]
 [14 16]]

Hasil Penjumlahan 2+A adalah: 
[[ 3  4]
 [ 9 10]]

Hasil Penjumlahan A+B adalah: 
[[ 4  7]
 [ 8 14]]

Hasil Pengurangan A-B adalah: 
[[-2 -3]
 [ 6  2]]

Determinan matriks A -6.0

Invers Matriks B: 
 [[ 0.46153846 -0.38461538]
 [-0.07692308  0.23076923]]

Transpose D: 
 [[ 0.46153846 -0.07692308]
 [-0.38461538  0.23076923]]


In [35]:
print('\nnilai eigen matriks A = ', np.linalg.eig(A)[0])
print('\nvektor eigen matriks A = \n', np.linalg.eig(A)[1])

print('\nnilai eigen matriks B = ', np.linalg.eig(B)[0])
print('\nvektor eigen matriks B = \n', np.linalg.eig(B)[1])


nilai eigen matriks A =  [-0.62347538  9.62347538]

vektor eigen matriks A = 
 [[-0.77640358 -0.22592838]
 [ 0.63023605 -0.97414391]]

nilai eigen matriks B =  [1.8074176 7.1925824]

vektor eigen matriks B = 
 [[-0.97271381 -0.76626443]
 [ 0.23200828 -0.64252535]]


In [39]:
#Tunjukkan bedanya di kodingan di bawah ini

B = np.array([[3, 5], [1,6]])
D = np.linalg.inv(B)

#perkalian antar entry di B dan D
J = B*D
print('perkalian antar entry di B dan D= ')
print(J)

#perkalian matriks B dan D
print('\nperkalian matriks B dan D = \n', np.matmul(B,D))

# Perkalian antar entry di B dan D merupakan perkalian setiap variabel dikalikan setiap elemen matriks

# Sedangkan, perkalian matriks B dan D merupakan hasil kali 2 matriks sesuai aturan perkalian matriks
# di mana baris dikali dengan kolom

perkalian antar entry di B dan D= 
[[ 1.38461538 -1.92307692]
 [-0.07692308  1.38461538]]

perkalian matriks B dan D = 
 [[1.00000000e+00 0.00000000e+00]
 [1.11022302e-16 1.00000000e+00]]


Apa perbedaan dari B*D (perkalian antar entry di B dan D) dan BD (perkalian matriks B dan D)? 

output jawaban: 

perkalian entry = [[ 1.38461538, -1.92307692], [-0.07692308,  1.38461538]]

perkalian matriks = [[ 1.00000000e+00, -1.11022302e-16],
       [ 1.11022302e-16,  1.00000000e+00]]
